ovos_plugin_common_play/ocp/media.py

```diff
@@ -545,9 +546,10 @@ def handle_media_state_change(self, message):
             state = MediaState(state)
         if not isinstance(state, MediaState):
             raise ValueError(f"Expected int or TrackState, but got: {state}")
-        if state == MediaState.END_OF_MEDIA:
-            # playback ended, allow next track to change metadata again
-            self.reset()
+        # Don't do anything. Let OCP manage this object's state
```

Comment:
this was added because the stream extractors dont replace old media, so if you send a request to play a uri only it will keep the title etc from previous track

the reset seems to be doing too much, but it is needed

Reply:
How would this request happen without going through OCP?

Reply:
OCP playback requests may contain full metadata or partial, metadata is then extracted from the uri automatically to fill gaps

if a playback request is missing info (title, artist, duration....) Now playing will keep that info when the next track comes up, the stream extractors will refuse to replace the now playing data (except uri) since they are a fallback, they assume the skill provided more accurate metadata than they are extracting

end result, Track 1 title with Track 2 playing

Reply:
note that the extractors only work at playback start, they have no way to know if the data in NowPlaying is old or not

if the reset is moved out of here it then needs to be done elsewhere before any playback request, that as more routes to happen so it seemed easier/safer to do it in here via bus events

Reply:
The call was moved [here](https://github.com/OpenVoiceOS/ovos-ocp-audio-plugin/pull/69/files#diff-2f0de75a91a2ac2d0725b6a5780231f4ad582527d9a3e89bad8489ac3387138cR246), is there another place I should add it? This handled all the paths I could find within OCP..

Reply:
i missed that change, guess that works!

please add a test for the specific scenario mentioned above if you haven't yet

Reply:
I didn't look at the diff lately and didn't realize how buried it is in other changes.